spec/ndx-pose.extensions.yaml

```diff
@@ -91,27 +120,86 @@ groups:
       dtype: text
       doc: Version string of the software tool used.
       required: false
-  - name: nodes
+  groups:
+  - neurodata_type_inc: PoseEstimationSeries
+    doc: Estimated position data for each body part.
+    quantity: '*'
+  links:
+  - target_type: Skeleton
+    doc: Layout of body part locations and connections.
+- neurodata_type_def: TrainingFrame
+  neurodata_type_inc: NWBDataInterface
+  default_name: TrainingFrame
+  doc: Group that holds ground-truth position data for all instances in a single frame.
+  attributes:
+  - name: annotator
     dtype: text
-    dims:
-    - num_body_parts
-    shape:
-    - null
-    doc: Array of body part names corresponding to the names of the PoseEstimationSeries
-      objects within this group.
+    doc: Name of annotator who labeled the TrainingFrame.
+    required: false
+  groups:
+  - neurodata_type_inc: Instance
+    doc: Position data for a single instance in a single training frame.
+    quantity: '*'
+  - name: source_video
+    doc: Path to original video file and frame used.
     quantity: '?'
-  - name: edges
+    attributes:
+    - name: path
+      dtype: text
+      doc: Path to original video file.
+      required: false
+    - name: frame_index
+      dtype: uint8
+      doc: Frame index of TrainingFrame in original video file.
+      required: false
+  - name: source_frame
+    neurodata_type_inc: Image
+    doc: Image frame used for training (stored either internally or externally).
+    quantity: '1'
```

Comment:
Note that currently the `Image` type supports only images stored internally. Also `quantity: 1` is fine to include, but it is the default when no quantity is provided, so we tend to omit it.

Reply:
From the past discussion, internally saving is the preferred approach since there could be thousands of external files otherwise.

However, maximal freedom for the user is also desired - would it be appropriate to extend the `Image` neurodata type in this extension to allow an external mode or should we alter that in the core instead?

Reply:
I created an issue https://github.com/NeurodataWithoutBorders/nwb-schema/issues/529 on the topic. This is part of a larger discussion over how we want to support storing external files in NWB.